src/cucumber/java/io.deepstream/ClientStepDefs.java

```diff
@@ -0,0 +1,74 @@
+package io.deepstream;
+
+import com.google.gson.JsonObject;
+import cucumber.api.java.en.Given;
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+import io.deepstream.constants.EndpointType;
+import io.deepstream.constants.Event;
+import org.junit.Assert;
+
+import java.util.Map;
+import java.util.Properties;
+
+public class ClientStepDefs {
+
+    private DeepstreamClient client;
+    Properties options = new Properties();
+    LoginStatus status = new LoginStatus();
+    DeepstreamException deepstreamException;
+
+    @Given("^the client is initialised$")
+    public void the_client_is_initialised() throws Throwable {
+        options.setProperty( "endpoint", EndpointType.TCP.name() );
+        client = new DeepstreamClient( "http://localhost:9696", options );
+    }
+
+    @Then("^the clients connection state is \"(.*?)\"$")
+    public void the_clients_connection_state_is( String arg1 ) throws Throwable {
+        Assert.assertEquals( arg1, client.getConnectionState().name() );
+    }
+
+    @When("^the client logs in with username \"(.*?)\" and password \"(.*?)\"")
+    public void The_client_logs_in_with_username_and_password( String username, String password ) throws Exception {
+        JsonObject authData = new JsonObject();
+        authData.addProperty( "password", password );
+        authData.addProperty( "username", username );
+        try {
+            client.login(authData, status);
+        } catch ( DeepstreamException ex ) {
+            deepstreamException = ex;

```

Comment:
Cool, that slipped my mind! Guess we should end up maintaing a folder full of exception types *shivers*